x-pack/plugins/ml/server/models/data_recognizer/data_recognizer.ts

```diff
@@ -124,6 +124,8 @@ export class DataRecognizer {
   private _resultsService: ReturnType<typeof resultsServiceProvider>;
   private _calculateModelMemoryLimit: ReturnType<typeof calculateModelMemoryLimitProvider>;
 
+  private _configCache: Config[] | null = null;
```

Comment:
Should we invalidate this cache if the saved object configs have been changed?

Reply:
As I mentioned in the description
>The modules on disk and in the saved objects will not change in the time frame that each DataRecognizer instance exist.

Each instance of the `DataRecognizer` exists for one endpoint call. The modules on disk will not change in that time frame. The modules in the saved object could theoretically change if someone installs a fleet module at the exact same time, but IMO it is very unlikely.

Reply:
Suggestion: It would be great if we could add some of the PR description and this explanation as comments to the code so the reasoning is also documented there.